src/Tests/Sending/When_routing_outgoingmessages_to_endpoints.cs

```diff
@@ -676,6 +678,176 @@ public Task Handle(BrokeredMessage brokeredMessage)
+            var fallback = new RuntimeNamespaceInfo("fallback", AzureServiceBusConnectionString.Fallback, mode: NamespaceMode.Passive);
+            var bytes = Encoding.UTF8.GetBytes("Whatever");
+            var batch = new Batch
+            {
+                Destinations = new TopologySection
+                {
+                    Entities = new List<EntityInfo>
+                    {
+                        new EntityInfo
+                        {
+                            Namespace = @namespace,
+                            Path = "MyQueue",
+                            Type = EntityType.Queue
+                        },
+                        new EntityInfo
+                        {
+                            Namespace = fallback,
+                            Path = "MyQueue",
+                            Type = EntityType.Queue
+                        }
+                    },
+                    Namespaces = new List<RuntimeNamespaceInfo>
+                    {
+                        @namespace,
+                        fallback
+                    }
+                },
+                RequiredDispatchConsistency = DispatchConsistency.Default,
+                Operations = new List<BatchedOperation>
+                {
+                    new BatchedOperation
+                    {
+                        Message = new OutgoingMessage("SomeId", new Dictionary<string, string>(), bytes),
+                        DeliveryConstraints = new List<DeliveryConstraint>()
+                    },
+                }
+            };
+
+            // perform the test
+            await router.RouteBatch(batch, null, DispatchConsistency.Default);
+
+            Assert.That(messageSenderCreator.Senders["primary"].NumberOfTimeInvoked, Is.EqualTo(1));
+            Assert.That(messageSenderCreator.Senders["fallback"].NumberOfTimeInvoked, Is.EqualTo(0));
+        }
+
+        [Test]
+        public async Task Can_route_via_fallback_namespace_no_ASB_required()
+        {
+            // default settings
+            var settings = new DefaultConfigurationValues().Apply(new SettingsHolder());
+
+            var messageSenderCreator = new FakeICreateMessageSenders("primary");
+            var clientLifecycleManager = new MessageSenderLifeCycleManager(messageSenderCreator, settings);
+            var router = new DefaultOutgoingBatchRouter(new DefaultBatchedOperationsToBrokeredMessagesConverter(settings), clientLifecycleManager, settings, new ThrowOnOversizedBrokeredMessages());
+
+            // setup the batch
+            var @namespace = new RuntimeNamespaceInfo("primary", AzureServiceBusConnectionString.Value);
+            var fallback = new RuntimeNamespaceInfo("fallback", AzureServiceBusConnectionString.Fallback, mode: NamespaceMode.Passive);
+            var bytes = Encoding.UTF8.GetBytes("Whatever");
+            var batch = new Batch
```

Comment:
Looks that this is duplicated between new tests. Maybe pull out to separate method.